components/range_cache_memory_engine/src/background.rs

```diff
@@ -956,34 +970,90 @@ impl Runnable for BackgroundRunner {
                                     }
                                     Err(e) => {
                                         error!("creating rocksdb iterator failed"; "cf" => cf, "err" => %e);
-                                        return false;
+                                        return None;
                                     }
                                 }
                             }
-                            true
+
+                            // gc the range
+                            let tso_timeout = std::cmp::min(gc_interval, Duration::from_secs(5));
+                            let now = match block_on_timeout(pd_client.get_tso(), tso_timeout) {
+                                Ok(Ok(ts)) => ts,
+                                err => {
+                                    error!(
+                                        "get timestamp failed, skip gc loaded range";
+                                        "timeout_duration" => ?tso_timeout,
+                                        "error" => ?err,
+                                    );
+                                    // Get timestamp fail so don't do gc.
+                                    return Some(0);
```

Comment:
If so => `now == Some(0)`, maybe all following steps can be skipped with directly returning `None` ?

Reply:
None means load is failed. Some(0) means the load success but does not get the tso for gc.

Reply:
got